ADC OFA drives LED BLUE

ik1xpv opened the pull request:
I use the pulse slope of ADC OFA to trigger a  retriggerable timer of 1000 ms. It switch the LED BLUE on and off.

I was not able to use the hardware complex GPIO mode in a simple way.
In cyu3gpio.h:
...
CY_U3P_GPIO_MODE_MEASURE_POS_ONCE,  /**< This mode is similar to the MEASURE_POS mode, except that only a
                                             single delay measurement is performed. **This mode cannot be directly
                                             selected**, and is used internally by the CyU3PGpioComplexMeasureOnce
                                             API. */
..
I made this modification in BBRF103, HF103, RX888
I tested it on BBRF103 and HF103. 
On RX888 it is compatible but I was not able to see where is the Blue LED :-)

In the code I modified the BBRF103 hardware detection code. Instead of using GPIO 54 I used GPIO 50 / 1k pull down 
as described  into file ..\HWSDRtable.h.

howard0su commented on SDDC_FX3/RunApplication.c:
```diff
@@ -69,46 +76,107 @@ ConfGPIOsimpleinput( uint8_t gpioid)
+		     if (rx888_OFA_Id(gpioId))
+			 {
+				 rx888_GpioSetOFA(true);
+				 CyU3PTimerModify (&OFATimer, OFA_PERIOD, 0);
```
move these two lines outr of the switch and add default branch to return.

howard0su commented on SDDC_FX3/RunApplication.c:
```diff
@@ -41,6 +41,12 @@ void *StackPtr[APP_THREADS];				// Stack allocated to each thread
 uint8_t HWconfig = NORADIO;       // Hardware config type BBRF103
 uint16_t FWconfig = 0x0102;    // Firmware rc1 ver 1.02
 
+#define BBRF103_PD 50  // BBRF103 pull down GPIO
```
can we move this to the hardware specific file?

howard0su commented on SDDC_FX3/makefile_win:
```diff
@@ -0,0 +1,85 @@
+## Copyright Cypress Semiconductor Corporation, 2010-2018,
```
why we need a windows specific Makefile?

howard0su replied:
can we modify the current one via check the enviroment?

howard0su commented on SDDC_FX3/rx888.c:
```diff
@@ -22,7 +23,7 @@ void rx888_GpioSet(uint32_t mdata)
     CyU3PGpioSetValue (GPIO_LED_RED, (mdata & LED_RED) == LED_RED);
     CyU3PGpioSetValue (GPIO_LED_YELLOW, (mdata & LED_YELLOW) == LED_YELLOW);
-    CyU3PGpioSetValue (GPIO_LED_BLUE, (mdata & LED_BLUE) == LED_BLUE);
+//  CyU3PGpioSetValue (GPIO_LED_BLUE, (mdata & LED_BLUE) == LED_BLUE); // used by  rx888_GpioSetOFA
```
please remove this line.

howard0su commented on SDDC_FX3/RunApplication.c:
```diff
@@ -69,46 +76,107 @@ ConfGPIOsimpleinput( uint8_t gpioid)
+			hf103_GpioSetOFA(false);
+			break;
+		case BBRF103:
+		    bbrf103_GpioSetOFA(false);
```
ident

howard0su commented:
RX888 will be hard to find those LEDs unless you disassembly it. looks good to me.

howard0su commented:
I am not sure if USB packet can carry one more bit somewhere when DMA the data to PC. OVA information will be useful when we trying to implement AGC.

ik1xpv commented:
@howard0su I will remove make file for windows. I was trying to solve with Franco the Windows environment compatibility with c99. 
I have the idea to count how many overflows we have in a second, It could be a better parameter for the AGC.
The host will have to pull the FX3 OFA info ?

howard0su commented:
sounds good to me. We can add another firmware API to get the information.

On Mon, Dec 14, 2020 at 3:58 PM Oscar Steila <notifications@github.com>
wrote:

> @howard0su <https://github.com/howard0su> I will remove make file for
> windows. I was trying to solve with Franco the Windows environment
> compatibility with c99.
> I have the idea to count how many overflows we have in a second, It could
> be a better parameter for the AGC.
> The host will have to pull the FX3 OFA info ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744252149>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF3GRHTDYHJCVYBTPQ4FH3SUXAR5ANCNFSM4UZVKTEQ>
> .
>


-- 
-Howard

ik1xpv commented:
Yes
I'm thinking of something like a filter to measure how many OFA events we
had in the past second TBD. Updated every 100 ms.
We could add an heartbeat  periodic API from the master that could carry
infos in both directions.
I converted the pull request to draft as it's evolving.

Oscar


On Mon, Dec 14, 2020 at 10:21 AM Howard Su <notifications@github.com> wrote:

> sounds good to me. We can add another firmware API to get the information.
>
> On Mon, Dec 14, 2020 at 3:58 PM Oscar Steila <notifications@github.com>
> wrote:
>
> > @howard0su <https://github.com/howard0su> I will remove make file for
> > windows. I was trying to solve with Franco the Windows environment
> > compatibility with c99.
> > I have the idea to count how many overflows we have in a second, It could
> > be a better parameter for the AGC.
> > The host will have to pull the FX3 OFA info ?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744252149>,
> or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAF3GRHTDYHJCVYBTPQ4FH3SUXAR5ANCNFSM4UZVKTEQ
> >
> > .
> >
>
>
> --
> -Howard
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744302574>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACLNBGHLNBGBQYX2FF7YSRTSUXKJBANCNFSM4UZVKTEQ>
> .
>

howard0su commented:
sounds good to me.

On Mon, Dec 14, 2020 at 5:36 PM Oscar Steila <notifications@github.com>
wrote:

> Yes
> I'm thinking of something like a filter to measure how many OFA events we
> had in the past second TBD. Updated every 100 ms.
> We could add an heartbeat periodic API from the master that could carry
> infos in both directions.
> I converted the pull request to draft as it's evolving.
>
> Oscar
>
>
> On Mon, Dec 14, 2020 at 10:21 AM Howard Su <notifications@github.com>
> wrote:
>
> > sounds good to me. We can add another firmware API to get the
> information.
> >
> > On Mon, Dec 14, 2020 at 3:58 PM Oscar Steila <notifications@github.com>
> > wrote:
> >
> > > @howard0su <https://github.com/howard0su> I will remove make file for
> > > windows. I was trying to solve with Franco the Windows environment
> > > compatibility with c99.
> > > I have the idea to count how many overflows we have in a second, It
> could
> > > be a better parameter for the AGC.
> > > The host will have to pull the FX3 OFA info ?
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744252149>,
> > or
> > > unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AAF3GRHTDYHJCVYBTPQ4FH3SUXAR5ANCNFSM4UZVKTEQ
> > >
> > > .
> > >
> >
> >
> > --
> > -Howard
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744302574>,
> or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ACLNBGHLNBGBQYX2FF7YSRTSUXKJBANCNFSM4UZVKTEQ
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ik1xpv/ExtIO_sddc/pull/96#issuecomment-744311740>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF3GRGPUXGEJPLMGAYRZ33SUXMBDANCNFSM4UZVKTEQ>
> .
>


-- 
-Howard

ik1xpv commented:
See ADC-overflow-drives-BLU-LED
